f600233d1b154d507b782f84604bcb12/update/sys_hub_action_type_definition_7232d6c92fe41110e58359a72799b6f0.xml

```diff
@@ -0,0 +1,1331 @@
+        <sys_updated_on>2022-07-06 02:05:54</sys_updated_on>
+        <value>35aa573fd7802200bdbaee5b5e610375</value>
+        <variable display_value="MID Application">f5e56d79b3101300176b051a16a8dce4</variable>
+    </sys_variable_value>
+    <sys_variable_value action="INSERT_OR_UPDATE">
+        <document>sys_hub_step_instance</document>
+        <document_key>4d88d6412f281110e58359a72799b6de</document_key>
+        <order>600</order>
+        <sys_class_name>sys_variable_value</sys_class_name>
+        <sys_created_by>jeff.segal-ctr@ccpo.mil</sys_created_by>
+        <sys_created_on>2022-07-06 02:05:54</sys_created_on>
+        <sys_id>dd88d6412f281110e58359a72799b6fa</sys_id>
+        <sys_mod_count>3</sys_mod_count>
+        <sys_updated_by>jeff.segal-ctr@ccpo.mil</sys_updated_by>
+        <sys_updated_on>2022-07-06 02:09:15</sys_updated_on>
+        <value>function addCostRecord(result, csp, organization, portfolio, taskOrder, clin, isActual) {
+  var gdt = new GlideDateTime();
+  gdt.setDisplayValue(result.month, "yyyy-MM");
+  
+  // Check to see if we already have a record for this Task Order / CLIN / Year-Month Tuple
+  var cost = new GlideRecord('x_g_dis_atat_costs');
+  cost.addQuery("task_order_number", taskOrder.task_order_number);
+  cost.addQuery("clin", clin.clin_number);
+  cost.addQuery("year_month", gdt.getDate());
+  cost.query();
+  
+  var isUpdate = false;
+  if (cost.hasNext()) {
+    cost.next();
+    isUpdate = true;       
+  }
+  else {
+    cost.newRecord();
+    isUpdate = false;       
+  }
+
+  cost.clin = clin.clin_number;
+  cost.csp = csp.sys_id;
+  cost.is_actual = isActual;
+  cost.organization = organization.sys_id;
+  cost.portfolio = portfolio.sys_id;
+  cost.service_agency = organization.service_agency;
+  cost.task_order_number = taskOrder.task_order_number;
+  cost.value = result.value;
+  cost.year_month = gdt;
+  
+  if (isUpdate) {
+    cost.update();
+  }
+  else {
+    cost.insert();
+  }
+  
+}
+
+function handleCostObject(costObject) {
+  var csp = new GlideRecord('x_g_dis_atat_cloud_service_provider');
+  csp.get("name", costObject.content.request.targetCsp.name);
+
+  var portfolio = new GlideRecord('x_g_dis_atat_portfolio');
+  portfolio.get("csp_portfolio_id", costObject.content.request.portfolioId);
+  
+  var organization = portfolio.acquisition_package.organization;
+    
+  var taskOrders = costObject.content.response.task_orders;
+  for (var i = 0; i &lt; taskOrders.length; i++) {
+    var taskOrder = taskOrders[i];
+    var clins = taskOrder.clins;
+    for (var j = 0; j &lt; clins.length; j++) {
+      var clin = clins[j];
+      
+      var actuals = clin.actual;
+      for (var k = 0; k &lt; actuals.length; k++) {
+        var actual = actuals[k];
+        var actual_results = actual.results;
+        for (var l = 0; l &lt; actual_results.length; l++) {
+          addCostRecord(actual_results[l], csp, organization, portfolio, taskOrder, clin, true);
```

Comment:
Was it possible to pass in an object with the required values, or were there issues with referencing values used with GlideRecord (e.g., `organization`, `portfolio`)?

Reply:
Yeah, that's what I was asking. I was just wondering if it was easier to pass in an object rather then each individual value. But thought maybe there were issues with accessing certain values that require a GlideRecord.